waiter/src/waiter/scheduler.clj

```diff
@@ -401,15 +400,25 @@
 (defn do-health-checks
   "Takes a map from service -> service instances and performs health checks in parallel. Returns a map of service -> service instances."
   [service->service-instances available? service-id->service-description-fn]
-  (let [service->service-instance-futures (start-health-checks service->service-instances available? service-id->service-description-fn)]
-    (loop [[[service {:keys [active-instances] :as instances}] & rest] (seq service->service-instance-futures)
-           service->service-instances' {}]
-      (if-not service
-        service->service-instances'
-        (let [active-instances (doall (map async/<!! active-instances))]
-          (recur rest (assoc service->service-instances'
-                        service
-                        (assoc instances :active-instances active-instances))))))))
+  (let [timeout-threshold-mins 2
+        timeout-channel (async/timeout (-> timeout-threshold-mins t/minutes t/in-millis))
+        slowness-monitor-promise (promise)]
+    (async/go
+      (async/<! timeout-channel)
+      (when (not (realized? slowness-monitor-promise))
+        (log/info "scheduler-syncer: health checks are taking longer than" timeout-threshold-mins "minute(s) to complete!")))
```

Comment:
This should be at least `warn`.

Reply:
Nitpick but I'm not a fan of exclamation points in error messages.

Reply:
Done.